File: python/fib/fib.pl
#!/usr/bin/env perl

# a simple stupid fib recursive function in perl

use strict;
use warnings;
use 5.010;
use Data::Dump qw(dump);

sub fib($);

sub fib($)
{
	my $n = shift;
	if ($n <= 2)  {
		return 1;
	}
	else  {
		return fib($n-1) + fib($n-2);
	}
}

my @arr = ();
my $upper = 12;
for (1 .. $upper)  {
	push @arr, fib($_);
}
dump(@arr);

File: python/fib/fib.py
#!/usr/bin/python3

import random

''' a simple and stupid fib function test
'''

def fib(n):
    if n<=2:
        return 1
    else:
        return fib(n-1)+fib(n-2)

# it is time/memory consuming while n is large
if __name__ == '__main__':
    for i in range(10):
        n = random.randint(2,100)
        print('fib({}) = {}'.format(n, fib(n)))
